StyleGuide.md

```diff
@@ -970,6 +975,111 @@ Use the following HTML and Liquid to include an image in a Markdown page:
+
+There are [basic](#basic-include-file-usage) and [advanced](#advanced-include-file-usage) methods for using include files. Use the basic method unless you are sure you need the advanced.
+
+<a name="basic-include-file-usage"></a>
+
+#### Basic include file usage
+
+The basic method for using an include file is:
+
+1. Find (or create) a block of content that you want to make appear on two or more different pages.
+1. Create a new file in the subdirectory of the `_includes` directory associated with the product you are working on. For example, if you are working on CockroachDB Self-Hosted v22.1, the directory will be `_includes/v22.1/`. If you are working on CockroachDB Dedicated, it will be `_includes/cockroachcloud`. If the include text is not version- or product-specific, put it in the `common` folder.
+1. In the pages where you want the included content to appear, add an `include` tag like one of the following: for CockroachDB Self-Hosted, `{% include {{page.version.version}}/some/shared-file.md %}`; for CockroachDB Dedicated, `{% include cockroachcloud/some/shared-file.md %}`.
+
+The contents of `shared-file.md` will now appear on all of the pages where you added the `include` tag.
+
+<a name="advanced-include-file-usage"></a>
+
+#### Advanced include file usage
+
+##### Different content depending on page name
+
+There may be cases where the content of the include file will need to vary slightly depending on what pages that content is being shared into. For example, while working on [cockroachdb/docs#12216](https://github.com/cockroachdb/docs/pull/12216),  I needed a way to:
+
+- Have text be a link on the [Known Limitations](https://www.cockroachlabs.com/docs/stable/known-limitations) page.
+- Have that same text _not_ be a link on the [Cost-Based Optimizer](https://www.cockroachlabs.com/docs/stable/cost-based-optimizer) page (since it would be a self-referring link).
+
+The way to do this in Jekyll (with its templating language Liquid) is to add the following content to an include file that is shared into both pages:
+
+    {% if page.name == "cost-based-optimizer.md" %} Locality-optimized search {% else %} [Locality-optimized search](cost-based-optimizer.html#locality-optimized-search-in-multi-region-clusters) {% endif %} only works for queries selecting a limited number of records (up to 10,000 unique keys). Also, It does not yet work with [`LIMIT`](limit-offset.html) clauses.
+
+The syntax is a little hard to read inline, but based on some experimenting it appears that it _must_ be written as one line so as not to introduce line breaks in the resulting text.
+
+Formatted for easier reading, the template code looks like:
+
+```
+{% if page.name == "cost-based-optimizer.md" %}  
+Locality-optimized search  
+{% else %}  
+[Locality-optimized search](cost-based-optimizer.html#locality-optimized-search-in-multi-region-clusters)  
+{% endif %}  
+```
+
+##### Different content depending on Cockroach Cloud or Self-Hosted
```

Comment:
Nit: "Cockroach Cloud" --> "CockroachCloud"

Reply:
Updated in the title and the first sentence of the paragraph!